### PR TITLE
Changes to return correct response code from the local HTTP proxy and…

### LINKED ISSUE #3210: Lantern doesn't include default list of blocked sites on initial startup

### LINKED ISSUE #3211: Consolidate lantern.yaml and cloud.yaml
This would unify scripts for releasing new versions.

### LINKED ISSUE #3212: Lantern incorrectly thinking chained servers are succeeding
...and using them exclusively in http.go as a result. The problem is that the local Lantern proxy will return 200 OK even on failures.

### PR DESCRIPTION
**Pull request by myleshorton**
… to generate configs the client side can properly use closes #3212, closes #3211, closes #3210

### CODE REVIEW

**Comment by myleshorton** [comment]
Hey @fffw I know you're supposed to be on vacation, but if you have a second would you mind taking a look at this one?

File: src/github.com/getlantern/flashlight/genconfig/genfallbacks.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+	"sync"
+	"text/template"
+
+	"github.com/getlantern/flashlight/client"
+	"github.com/getlantern/golog"
+	"github.com/getlantern/yaml"
+)
+
+const (
+	numberOfWorkers = 50
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/genconfig/genfallbacks.go]: These two constants are no longer used.

File: src/github.com/getlantern/flashlight/genconfig/genfallbacks.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+	ftVersionFile   = `https://raw.githubusercontent.com/firetweet/downloads/master/version.txt`
+)
+
+var (
+	help            = flag.Bool("help", false, "Get usage help")
+	masqueradesFile = flag.String("masquerades", "", "Path to file containing list of pasquerades to use, with one space-separated 'ip domain' pair per line (e.g. masquerades.txt)")
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/genconfig/genfallbacks.go]: Also these variables.

File: src/github.com/getlantern/flashlight/genconfig/genfallbacks.go (hunk 1 above)
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/genconfig/genfallbacks.go]: I think they are still used right @fffw?

File: src/github.com/getlantern/flashlight/genconfig/genfallbacks.go (hunk 2 above)
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/genconfig/genfallbacks.go]: Wait I think those are still also used right? The masqueradesFile and help?

File: src/github.com/getlantern/flashlight/genconfig/genfallbacks.go (hunk 1 above)
Comment (fffw) [inline src/github.com/getlantern/flashlight/genconfig/genfallbacks.go]: I think they are used by `genconfig.go` but not `genfallbacks.go`? Nothing important anyway, it's just a command line tool.

File: src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl [hunk 3]
```diff
@@ -7,7 +7,7 @@ client:
   frontedservers: []
   chainedservers: {{range .fallbacks}}
     fallback-{{.ip}}:
-      addr: {{.ip}}:443
+      addr: {{.ip}}
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl]: What's the intention of this change?

Comment (myleshorton) [inline src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl]: Basically the ChainedServerInfo Addr field includes the port, which wasn't the case with the previous IP setting @fffw.

Comment (fffw) [inline src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl]: So `fallback-{{.ip}}:` will expand to `fallback-x.x.x.x:443:`? The extra colon seems invalid.

File: src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl [hunk 4]
```diff
@@ -20,6 +20,9 @@ client:
     - domain: {{.Domain}}
       ipaddress: {{.IpAddress}}{{end}}
 proxiedsites:
+  delta:
+    additions: []
+    deletions: []
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl]: This change just makes the generated yaml file actually work on the client side -- without this the client side won't load.

File: src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl (hunk 3 above)
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl]: Oh good point @fffw. It seems to work for some reason, but let me clarify.

Comment (myleshorton) [inline src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl]: Or sorry let me make sure rather =).

Comment (myleshorton) [inline src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl]: Yeah it works for whatever reason @fffw -- maybe it looks for colon followed by some whitespace?

Comment (fffw) [inline src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl]: Maybe wrap above two line with quotes? That will make it more clear.

Comment (fffw) [inline src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl]: Ok http://www.yaml.org/spec/1.2/spec.html#id2759963 says that `Mappings use a colon and space (“: ”) to mark each key: value pair`, so it's legal yaml.

Comment (myleshorton) [inline src/github.com/getlantern/flashlight/genconfig/cloud.yaml.tmpl]: Awesome @fffw thanks!

**Comment by fffw** [comment]
in `util/http.go#L69`, `chainedFetcher.Do` should check the response status code

**Comment by fffw** [comment]
Ok @myleshorton once the above bug fixed, everything should be ok to me.

**Comment by myleshorton** [comment]
OK thanks @fffw I'm on it!

**Comment by myleshorton** [comment]
OK got that in @fffw!

**Pull request merged**